packages/react/src/components/LocaleModal/LocaleModalCountries.js

```diff
@@ -21,19 +21,19 @@ const { prefix } = settings;
  * @param {object} props props object
  * @param {object} props.regionList object of country and language codes
  * @param {Function} props.setClearResults set flag to determine whether to reset the filtered results
+ * @param {string} props.currentRegion current region
  * @returns {*} LocaleModal component
  */
 const LocaleModalCountries = ({
   regionList,
   setClearResults,
+  currentRegion,
   ...modalLabels
 }) => {
   const localList = useRef(null);
   useEffect(() => {
-    if (open) {
-      localList.current.scrollTop = 0;
-    }
-  }, [open]);
+    localList.current.scrollTop = 0;
+  }, [currentRegion, regionList]);
```

Comment:
I think it still make sense to reset the scroll position only when the modal is open:

```suggestion
    if (open) {
      localList.current.scrollTop = 0;
    }
  }, [currentRegion, regionList, open]);
```

Reply:
This props is being set on the LocaleModal component which contains LocaleModalCountries and LocaleModalRegions, so I need to send the open props to LocaleModalCountries and every time LocalModal is loaded/opened LocaleModalCountries is too, however LocaleModal always keeps loaded/opened so LocaleModalCountries too so the open dependency will never change, that's why we put currentRegion and regionList as dependency, because every time region is selected reset the scroll position. @asudoh

Reply:
Great point, thank you for elaborating. And my apologies I missed that `open` ref there seems to have never worked before-your-change.

One way we can address this is making `localeList` React ref in `<LocaleModalCountries>` a React prop, that is passed along from `<LocaleModal>`. In this way, `<LocaleModal>` can grab the ref and change the `.scrollTop` when needed. I imagine `<LocalModal>` has better context of when to reset the scroll position and thus I think "resetting scroll position" logic more robust.

Reply:
@diegomarcuz @asudoh This is targeted for `v1.10.0`, think we can merge as-is then get in another change in a separate PR post code-freeze?

Reply:
@jeffchew Sounds good to me.